mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java

```diff
@@ -117,21 +117,22 @@ public static int getInitParameterAsStrictlyPositiveInteger(String condition, in
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition) throws MessagingException {
-        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.<String>absent());
+        return getInitParameterAsStrictlyPositiveInteger(condition, Optional.empty());
     }
 
     public static int getInitParameterAsStrictlyPositiveInteger(String condition, Optional<String> defaultValue) throws MessagingException {
-        Optional<String> value = Optional.fromNullable(condition)
-            .or(defaultValue);
+        String value = Optional.ofNullable(condition)
+            .filter(Strings::isNullOrEmpty)
+            .orElse(defaultValue.orElse(null));
 
-        if (Strings.isNullOrEmpty(value.orNull())) {
+        if (Strings.isNullOrEmpty(value)) {
```

Comment:
this is useless now that you filter, but then you don't get the error message, so please keep the old behavior

Reply:
No, it's not useless if defaultValue is null also